app/controllers/dances_controller.rb

```diff
@@ -15,9 +17,11 @@ def show
 
   def new
     @dance = Dance.new
+    @dance_direction = DanceDirection.new
   end
 
   def create
+    # NEED TO ACCOUNT FOR THE EMPTY "" STRINGS BEING ENTERED INTO THE DATABASE IN PLACE OF NIL
     @dance = Dance.new(dance_params.merge(user_id: current_user.id.to_s))
```

Comment:
Got this trick from this post: https://stackoverflow.com/questions/7202319/rails-force-empty-string-to-null-in-the-database


A poster recommends, https://github.com/rubiety/nilify_blanks which may be worth looking into